sympy/simplify/cse_main.py

```diff
@@ -248,7 +248,7 @@ def _match_common_args(Func, funcs):
     # split muls into commutative
     comutative_muls = set()
     for m in muls:
-        c, nc = m.args_cnc(cset=True)
+        c, nc = m.args_cnc(cset=True, warn=False)
```

Comment:
I thought the point of `warn` here was that `cset` removes duplicate factors, since they are put in a set. I guess that's not an issue here because we're just accumulating subexpressions?

Reply:
You are right, it should be `cset=False`